bin/create-eslint-config.js

```diff
@@ -88,7 +93,39 @@ if (pkg.eslintConfig) {
   }
 }
 
-// 2. Check Vue
+// 2. Config format
+let configFormat
+try { 
+  const eslintVersion = requireInCwd('eslint/package.json').version
+  console.info(dim(`Detected ESLint version: ${eslintVersion}`))
+  const [major, minor] = eslintVersion.split('.')
+  if (parseInt(major) >= 9) {
+    configFormat = 'flat'
+  } else if (parseInt(major) === 8 && parseInt(minor) >= 21) {
```

Comment:
The [user guide](https://eslint.vuejs.org/user-guide/) for `eslint-plugin-vue` mentions that flat config files "can be used starting from ESLint v8.57.0", so this logic should probably match that version number.